src/jit/codegenlinear.cpp

```diff
@@ -808,12 +815,22 @@ void CodeGen::genSpillVar(GenTree* tree)
         }
     }
 
+    // Why is changing reg if spill was not needed?
     tree->gtFlags &= ~GTF_SPILL;
     varDsc->lvRegNum = REG_STK;
     if (varTypeIsMultiReg(tree))
     {
         varDsc->lvOtherReg = REG_STK;
     }
+
+    if (needsSpill)
+    {
+        // We need this after "lvRegNum" has change because now we are sure that varDsc->lvIsInReg() is false.
+        // "SiVarLoc" constructor uses the "LclVarDsc" of the variable.
```

Comment:
This comment seems a little strange here because there's no clear connection between the following and `SiVarLoc` or its constructor. I would just omit this line.